cpp/lib/src/MarcRecord.cc

```diff
@@ -0,0 +1,296 @@
+#include "MarcRecord.h"
+#include "util.h"
+#include "MarcReader.h"
+
+std::string MarcRecord::getFieldData(const std::string &tag) const {
+    return getFieldData(getFieldIndex(tag));
+}
+
+
+std::string MarcRecord::getFieldData(const size_t index) const {
+    if (index == MarcRecord::FIELD_NOT_FOUND || directory_entries_.cbegin() + index >= directory_entries_.cend())
+        return "";
+    const DirectoryEntry &entry(directory_entries_[index]);
+    if (not entry.hasCache()) {
+        entry.setCache(std::string(raw_data_, entry.getFieldOffset(), entry.getFieldLength() - 1));
+    }
+    return entry.getCache();
+}
+
+
+Subfields MarcRecord::getSubfields(const std::string &tag) const {
+    return getSubfields(getFieldIndex(tag));
+}
+
+
+Subfields MarcRecord::getSubfields(const size_t index) const {
+    if (index == MarcRecord::FIELD_NOT_FOUND || directory_entries_.cbegin() + index >= directory_entries_.cend())
```

Comment:
Also, I am not sure that I like this strategy of being able to get things that don't exist.